diff -ruN fuego-1.0-original-with-mainfunction-pipeargs-patches/fuegomain/FuegoMainUtil.cpp fuego-1.0-new/fuegomain/FuegoMainUtil.cpp
--- fuego-1.0-original-with-mainfunction-pipeargs-patches/fuegomain/FuegoMainUtil.cpp	2011-03-24 21:36:01.000000000 +0100
+++ fuego-1.0-new/fuegomain/FuegoMainUtil.cpp	2011-03-24 21:38:25.000000000 +0100
@@ -119,7 +119,14 @@
     out << "Usage: fuego [options] [input files]\n" << desc << "\n";
 }
 
-void ParseOptions(int argc, char** argv)
+/** Returns normally if no error occurred. Returns true if program should
+    continue running, false if program should be terminated (e.g.
+    because help was requested). In the latter case, exit code 0 should
+    be used.
+
+    Throws an exception of undefined type if any error occurs. The
+    program should be terminated in response using an exit code != 0. */
+bool ParseOptions(int argc, char** argv)
 {
     po::options_description normalOptions("Options");
     normalOptions.add_options()
@@ -172,7 +179,10 @@
         throw;
     }
     if (vm.count("help"))
+    {
         Help(normalOptions, cout);
+        return false;
+    }
     if (vm.count("nobook"))
         g_noBook = true;
     if (vm.count("nohandicap"))
@@ -238,7 +248,9 @@
         g_programDir = GetProgramDir(argv[0]);
         try
         {
-            ParseOptions(argc, argv);
+            bool okAndContinue = ParseOptions(argc, argv);
+            if (! okAndContinue)
+                return 0;
         }
         catch (const SgException& e)
         {
